### DISCUSSION TITLE
how to run xcode with rosetta ?

### DISCUSSION THREAD

**Discussion by sad-adnan** [codemagic-ci-cd/codemagic-docs#2574] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

in my flutter project i have googlemaps, which works perfectly on physical devices but in `flutter build ios --debug --simulator` it thorws below error 

```
Building (app bundle) for simulator (ios)...
Updating project for Xcode compatibility.
Upgrading project.pbxproj
Upgrading Runner.xcscheme
Running pod install...                                           2,721ms
Running Xcode build...                                          
Xcode build done.                                           29.8s
Failed to build iOS app
Error (Xcode): Undefined symbol: _OBJC_CLASS_$_FLTGoogleMapsPlugin

Error (Xcode): Linker command failed with exit code 1 (use -v to see invocation)

Encountered error while building for simulator.


Build failed :|
Failed to build for iOS
```

i did some research and found that the google maps plugin is not switched to m1 chip or something and others suggested that i run it through rosetta , but i can't run the xcode with rosetta as there is no option , is there any workaround ?

### Expected results

i should be able to build `flutter build ios --debug --simulator` without any issues , the project should be build for simulator even in M1 Mac Mini even the Google Map plugin exists in the project.

### Actual results

it gives below error:

```
Building (app bundle) for simulator (ios)...
Updating project for Xcode compatibility.
Upgrading project.pbxproj
Upgrading Runner.xcscheme
Running pod install...                                           2,721ms
Running Xcode build...                                          
Xcode build done.                                           29.8s
Failed to build iOS app
Error (Xcode): Undefined symbol: _OBJC_CLASS_$_FLTGoogleMapsPlugin

Error (Xcode): Linker command failed with exit code 1 (use -v to see invocation)

Encountered error while building for simulator.


Build failed :|
Failed to build for iOS
```

**Comment by himesh-cm** [comment]
Hey @sad-adnan 
Hey, the issue seems to be with Google maps Plugin
Please check if this [GH](https://github.com/flutter/flutter/issues/133866#issuecomment-1702992460) link for possible solutions.

**Comment by himesh-cm** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
